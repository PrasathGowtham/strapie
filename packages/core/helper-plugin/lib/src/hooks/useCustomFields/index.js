/**
 *
 * useCustomFields
 *
 */

import { useContext, useRef } from 'react';
import CustomFieldsContext from '../../contexts/CustomFieldsContext';

const useCustomFields = () => {
  const customFields = useContext(CustomFieldsContext);
  // Use a ref so we can safely add the custom fields to a hook dependencies array
  const customFieldsRef = useRef(customFields);

  return customFieldsRef.current;
};

export default useCustomFields;
