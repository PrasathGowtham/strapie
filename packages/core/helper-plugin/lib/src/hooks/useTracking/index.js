import { useContext, useRef } from 'react';
import axios from 'axios';
import TrackingContext from '../../contexts/TrackingContext';
import useAppInfos from '../useAppInfos';

const useTracking = () => {
  const trackRef = useRef();
  const { uuid, telemetryProperties } = useContext(TrackingContext);
  const appInfo = useAppInfos();

  trackRef.current = async (event, properties) => {
    if (uuid && !window.strapi.telemetryDisabled) {
      try {
        await axios.post('https://analytics.strapi.io/track', {
          event,
          properties: {
            ...telemetryProperties,
            ...properties,
            projectType: window.strapi.projectType,
            environment: appInfo.currentEnvironment,
          },
          uuid,
        });
      } catch (err) {
        // Silent
      }
    }
  };

  return { trackUsage: trackRef.current };
};

export default useTracking;
