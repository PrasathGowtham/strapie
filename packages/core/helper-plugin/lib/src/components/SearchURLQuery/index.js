import React, { useLayoutEffect, useState, useRef } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';
import SearchIcon from '@strapi/icons/Search';
import { Searchbar, SearchForm } from '@strapi/design-system/Searchbar';
import { IconButton } from '@strapi/design-system/IconButton';
import { Icon } from '@strapi/design-system/Icon';
import useQueryParams from '../../hooks/useQueryParams';
import useTracking from '../../hooks/useTracking';

const SearchURLQuery = ({ label, placeholder, trackedEvent, trackedEventDetails }) => {
  const wrapperRef = useRef(null);
  const iconButtonRef = useRef(null);

  const [{ query }, setQuery] = useQueryParams();
  const [value, setValue] = useState(query?._q || '');
  const [isOpen, setIsOpen] = useState(!!value);
  const { formatMessage } = useIntl();
  const { trackUsage } = useTracking();

  const handleToggle = () => setIsOpen((prev) => !prev);

  useLayoutEffect(() => {
    if (isOpen) {
      setTimeout(() => {
        wrapperRef.current.querySelector('input').focus();
      }, 0);
    }
  }, [isOpen]);

  const handleClear = () => {
    setValue('');
    setQuery({ _q: '' }, 'remove');
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    if (value) {
      if (trackedEvent) {
        trackUsage(trackedEvent, trackedEventDetails);
      }
      setQuery({ _q: encodeURIComponent(value), page: 1 });
    } else {
      handleToggle();
      setQuery({ _q: '' }, 'remove');
    }
  };

  if (isOpen) {
    return (
      <div ref={wrapperRef}>
        <SearchForm onSubmit={handleSubmit}>
          <Searchbar
            name="search"
            onChange={({ target: { value } }) => setValue(value)}
            value={value}
            clearLabel={formatMessage({ id: 'clearLabel', defaultMessage: 'Clear' })}
            onClear={handleClear}
            size="S"
            placeholder={placeholder}
          >
            {label}
          </Searchbar>
        </SearchForm>
      </div>
    );
  }

  return (
    <IconButton
      ref={iconButtonRef}
      icon={<Icon as={SearchIcon} color="neutral800" />}
      label={formatMessage({ id: 'global.search', defaultMessage: 'Search' })}
      onClick={handleToggle}
    />
  );
};

SearchURLQuery.defaultProps = {
  placeholder: undefined,
  trackedEventDetails: undefined,
  trackedEvent: null,
};

SearchURLQuery.propTypes = {
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  trackedEventDetails: PropTypes.object,
  trackedEvent: PropTypes.string,
};

export default SearchURLQuery;
