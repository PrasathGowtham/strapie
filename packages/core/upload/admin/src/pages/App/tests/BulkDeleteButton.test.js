import React from 'react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { QueryClientProvider, QueryClient } from 'react-query';
import { render, fireEvent, waitFor, act } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { IntlProvider } from 'react-intl';

import { BulkDeleteButton } from '../components/BulkDeleteButton';
import { useBulkRemove } from '../../../hooks/useBulkRemove';

jest.mock('../../../hooks/useBulkRemove');

const setup = (
  props = {
    selected: [],
    onSuccess: jest.fn(),
  }
) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
        refetchOnWindowFocus: false,
      },
    },
  });

  return render(
    <QueryClientProvider client={queryClient}>
      <ThemeProvider theme={lightTheme}>
        <MemoryRouter>
          <IntlProvider locale="en">
            <BulkDeleteButton {...props} />
          </IntlProvider>
        </MemoryRouter>
      </ThemeProvider>
    </QueryClientProvider>
  );
};

describe('BulkDeleteButton', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  test('opens confirm dialog before the API call', () => {
    const onSuccessSpy = jest.fn();
    const { getByText } = setup({
      onSuccess: onSuccessSpy,
      selected: [{ type: 'asset' }],
    });
    const removeSpy = jest.fn().mockResolvedValueOnce({});

    useBulkRemove.mockReturnValueOnce({
      isLoading: false,
      error: null,
      remove: removeSpy,
    });

    act(() => {
      fireEvent.click(getByText('Delete'));
    });

    expect(getByText('Are you sure you want to delete this?')).toBeInTheDocument();

    act(() => {
      fireEvent.click(getByText('Confirm'));
    });

    expect(removeSpy).toBeCalledWith([{ type: 'asset' }]);
    waitFor(() => expect(onSuccessSpy).toBeCalledTimes(1));
  });
});
