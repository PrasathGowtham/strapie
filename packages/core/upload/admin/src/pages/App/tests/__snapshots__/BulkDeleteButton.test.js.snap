// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BulkDeleteButton renders 1`] = `
.c7 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c6 {
  font-size: 0.75rem;
  line-height: 1.33;
  font-weight: 600;
  line-height: 1.14;
  color: #32324d;
}

.c3 {
  padding-right: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  padding: 8px;
  border-radius: 4px;
  background: #ffffff;
  border: 1px solid #dcdce4;
  position: relative;
  outline: none;
}

.c0 svg {
  height: 12px;
  width: 12px;
}

.c0 svg > g,
.c0 svg path {
  fill: #ffffff;
}

.c0[aria-disabled='true'] {
  pointer-events: none;
}

.c0:after {
  -webkit-transition-property: all;
  transition-property: all;
  -webkit-transition-duration: 0.2s;
  transition-duration: 0.2s;
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -4px;
  bottom: -4px;
  left: -4px;
  right: -4px;
  border: 2px solid transparent;
}

.c0:focus-visible {
  outline: none;
}

.c0:focus-visible:after {
  border-radius: 8px;
  content: '';
  position: absolute;
  top: -5px;
  bottom: -5px;
  left: -5px;
  right: -5px;
  border: 2px solid #4945ff;
}

.c4 {
  height: 100%;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #4945ff;
  border: 1px solid #4945ff;
  height: 2rem;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid #f5c0b8;
  background: #fcecea;
}

.c1 .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 .c5 {
  color: #ffffff;
}

.c1[aria-disabled='true'] {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c1[aria-disabled='true'] .c5 {
  color: #666687;
}

.c1[aria-disabled='true'] svg > g,.c1[aria-disabled='true'] svg path {
  fill: #666687;
}

.c1[aria-disabled='true']:active {
  border: 1px solid #dcdce4;
  background: #eaeaef;
}

.c1[aria-disabled='true']:active .c5 {
  color: #666687;
}

.c1[aria-disabled='true']:active svg > g,.c1[aria-disabled='true']:active svg path {
  fill: #666687;
}

.c1:hover {
  background-color: #ffffff;
}

.c1:active {
  background-color: #ffffff;
  border: 1px solid #d02b20;
}

.c1:active .c5 {
  color: #d02b20;
}

.c1:active svg > g,
.c1:active svg path {
  fill: #d02b20;
}

.c1 .c5 {
  color: #b72b1a;
}

.c1 svg > g,
.c1 svg path {
  fill: #b72b1a;
}

<div>
  <button
    aria-disabled="false"
    class="c0 c1"
    type="button"
  >
    <div
      aria-hidden="true"
      class="c2 c3 c4"
    >
      <svg
        fill="none"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3.236 6.149a.2.2 0 00-.197.233L6 24h12l2.96-17.618a.2.2 0 00-.196-.233H3.236zM21.8 1.983c.11 0 .2.09.2.2v1.584a.2.2 0 01-.2.2H2.2a.2.2 0 01-.2-.2V2.183c0-.11.09-.2.2-.2h5.511c.9 0 1.631-1.09 1.631-1.983h5.316c0 .894.73 1.983 1.631 1.983H21.8z"
          fill="#32324D"
        />
      </svg>
    </div>
    <span
      class="c5 c6"
    >
      Delete
    </span>
  </button>
  <div
    class="c7"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
